pynest/nest/lib/hl_api_simulation.py

```diff
@@ -366,27 +377,37 @@ def SetKernelStatus(params):
         are counted, not spikes generated by devices such as ``poisson_generator``.
 
 
-    **Miscellaneous**
-
-    Parameters
-    ----------
-
-    dict_miss_is_error : bool
-        Whether missed dictionary entries are treated as errors
-    keep_source_table : bool
-        Whether to keep source table after connection setup is complete
-
     See Also
     --------
 
     GetKernelStatus
 
     """
-
+    for key in params.keys():
+        readonly = _sks_params.get(key)
+        if readonly is None:
+            # If the parameter is not in the docstring
+            raise KeyError(f'`{key}` is not a valid kernel parameter, '
+                           'valid parameters are: '
+                           ', '.join(f"'{p}'" for p in _sks_params.keys()))
```

Comment:
There should be a `+` here for correct operator precedence, otherwise the whole string gets joined:

```python
            raise KeyError(f'`{key}` is not a valid kernel parameter, '
                           'valid parameters are: '
                           + ', '.join(f"'{p}'" for p in _sks_params.keys()))
```

The way `KeyError` is displayed by Python is to `repr(exc.args[0])`, to show which key you indexed that errors, and our message is shown as a string (so there's quotes around it). 

Here we want to show an error message about incorrect argument values so I think a `ValueError` is a better fit for incorrect argument values (see [here](https://docs.python.org/3/library/exceptions.html#ValueError) and the note [here](https://docs.python.org/3/library/exceptions.html#TypeError).

Reply:
I have changed the error type

Reply:
I believe that also the joining error is now solved